src/main/java/lib/motors/SparkMaxSim.java

```diff
@@ -0,0 +1,100 @@
+package lib.motors;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.REVLibError;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+
+public class SparkMaxSim {
+
+    private final DCMotorSim motorSim;
```

Comment:
This member should be of type [`LinearSystemSim`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/simulation/LinearSystemSim.html) because there are three useful more-specific subclasses ([`ElevatorSim`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/simulation/ElevatorSim.html), [`FlywheelSim`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/simulation/FlywheelSim.html), [`SingleJointedArmSim`](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/simulation/SingleJointedArmSim.html)). The specific sim object should be supplied as a ctor parameter. Applies to the `TalonFXSim` too.